lib/rubocop/cop/style/nested_function_definition.rb

```diff
@@ -0,0 +1,19 @@
+# encoding: utf-8
+
+module RuboCop
+  module Cop
+    module Style
+      # This cop checks for nested function definitions.
+      class NestedFunctionDefinition < Cop
```

Comment:
Include the `OnMethodDef` mixin to support singleton method definitions as well.